src/en/snowmtl/src/eu/kanade/tachiyomi/extension/en/snowmtl/Snowmtl.kt

```diff
@@ -0,0 +1,480 @@
+                }
+            }
+
+            if (currentLine.isNotEmpty()) {
+                lines.add(currentLine.toString())
+            }
+
+            return lines
+        }
+    }
+
+    // =============================== Filters ================================
+
+    override fun getFilterList(): FilterList {
+        val filters = mutableListOf<Filter<*>>(
+            SelectionList("Sort", sortByList),
+            Filter.Separator(),
+            GenreList(title = "Genres", genres = genreList),
+        )
+
+        return FilterList(filters)
+    }
+
+    private var genreList: List<Genre> = listOf(
+        Genre("Action"),
+        Genre("Adult"),
+        Genre("Adventure"),
+        Genre("Comedy"),
+        Genre("Drama"),
+        Genre("Ecchi"),
+        Genre("Fantasy"),
+        Genre("Gender Bender"),
+        Genre("Harem"),
+        Genre("Historical"),
+        Genre("Horror"),
+        Genre("Josei"),
+        Genre("Lolicon"),
+        Genre("Martial Arts"),
+        Genre("Mature"),
+        Genre("Mecha"),
+        Genre("Mystery"),
+        Genre("Psychological"),
+        Genre("Romance"),
+        Genre("School Life"),
+        Genre("Sci-fi"),
+        Genre("Seinen"),
+        Genre("Shoujo"),
+        Genre("Shoujo Ai"),
+        Genre("Shounen"),
+        Genre("Shounen Ai"),
+        Genre("Slice of Life"),
+        Genre("Smut"),
+        Genre("Sports"),
+        Genre("Supernatural"),
+        Genre("Tragedy"),
+        Genre("Yaoi"),
+        Genre("Yuri"),
+    )
+
+    private val sortByList = listOf(
+        Option("All"),
+        Option("Most Views", "views"),
+        Option("Most Recent", "recent"),
+    ).map { it.copy(query = "sort_by") }
+
+    private data class Option(val name: String = "", val value: String = "", val query: String = "")
+
+    private open class SelectionList(displayName: String, private val vals: List<Option>, state: Int = 0) :
+        Filter.Select<String>(displayName, vals.map { it.name }.toTypedArray(), state) {
+        fun selected() = vals[state]
+    }
+
+    private class GenreList(title: String, genres: List<Genre>) :
+        Filter.Group<GenreCheckBox>(title, genres.map { GenreCheckBox(it.name, it.id) })
+
+    class GenreCheckBox(name: String, val id: String = name) : Filter.CheckBox(name)
+
+    class Genre(val name: String, val id: String = name)
+
+    companion object {
+        const val scaledDensity = 1.5f
+        const val PREFIX_SEARCH = "id:"
+        val MEDIA_TYPE = "image/png".toMediaType()
+        private val dateFormat: SimpleDateFormat = SimpleDateFormat("dd MMMM yyyy", Locale.US)
+    }
+}
+
+fun Paint.getCharWidth(): Float {
+    val text = "A"
+    val fontWidth = FloatArray(1)
+    getTextWidths(text.first().toString(), fontWidth)
+    return fontWidth.first()
+}
+
+fun Paint.getCharHeight(): Float {
+    val text = "A"
+    val bounds = Rect()
+    getTextBounds(text, 0, text.length, bounds)
+    return bounds.height().toFloat()
+}
```

Comment:
I need the dimensions (width x height) of the char to carry out the following process (I've added comments to the functions)

### Given

* Caption: Manga is Good
* Dialog Box:
```
   16px
+--------+
|        | 
|        | 
|        |
|        | 20px
|        |
|        |
+--------+

```

* Char width = 2px
* Char heigth = 3px


### When

Max char per line  : 16px / 2px  = 8 
Max number of lines: 20px / 3px    =  6

Breaking text into lines

```
  2px   2px   2px   2px   2px   2px   2px   2px  
|  M  |  a  |  n  |  g  |  a  |     |  i  |  s  | 3px
|  G  |  o  |  o  |  d  |     |     |     |     | 3px
|     |     |     |     |     |     |     |     | 3px
|     |     |     |     |     |     |     |     | 3px
|     |     |     |     |     |     |     |     | 3px
|     |     |     |     |     |     |     |     | 3px

```

### Then

Dialog box

```
   16px
+--------+
|Manga is| 
|Good    | 
|        |
|        | 20px
|        |
|        |
+--------+
```

Reply:
That only works for monospace fonts, otherwise it will look like:
>    16px
> +--------+
> |Manga is| 
> |Good    | 
> |        |
> |        | 20px
> |        |
> |        |
> +--------+


Someone made a `Canvas.drawMultilineText()` function that uses `StaticLayout` under the hood- https://medium.com/over-engineering/drawing-multiline-text-to-canvas-on-android-9b98f0bfa16a

https://stackoverflow.com/a/41779935 might also be a useful reference

If you can't use it directly, you could treat that function as a black box and decrease the font size until it fits the text.